README.md
# rental_store_maeganavant


history.txt
Item..., Days..., Price..., Total.....
Mardi Gras Mask, 2, 3.50, 7.49
Outdoor Wooden Dance Floor, 3, 50.00, 160.5
Mardi Gras Mask, 2, 3.50, 7.49
Mardi Gras Mask, 7, 3.50, 26.215
Mardi Gras Mask, 1, 3.50, 3.745
Mardi Gras Mask, 1, 3.50, 3.745
Can Lights, 5, 3.50, 18.725
Mardi Gras Mask, 1, 3.50, 3.745
Mardi Gras Mask, 1, 100, 107.0
Mardi Gras Mask, 12, 3.50, 44.94
Mardi Gras Street Signs, 9, 5.00, 48.15
Mardi Gras Mask, 1, 3.50, 3.745
Mardi Gras Street Signs, 34, 5.00, 181.9
Mardi Gras Mask,1,3.50,3.745
Mardi Gras Street Signs,11,5.001,58.85
Outdoor Wooden Dance Floor,2,50.00,107.0
Mardi Gras Mask, 1, 3.50, 3.745 
Mardi Gras Mask, 1, 3.50, 3.745 
Outdoor Wooden Dance Floor, 2, 50.00, 107.0 
Mardi Gras Mask, 1, 3.50, 3.745 
Building Props, 3, 3.50, 11.235 
Mardi Gras Street Signs, 12, 5.00, 64.2 
Mardi Gras Mask, 1, 3.50, 3.745 
Mardi Gras Mask, 1, 3.50, 3.745 
Mardi Gras Mask, 12, 3.50, 44.94 
Mardi Gras Mask, 1, 3.50, 3.745 
Mardi Gras Mask, 121, 3.50, 453.145 
Mardi Gras Mask, 123, 3.50, 460.635 
Mardi Gras Mask, 11, 3.50, 41.195 
Mardi Gras Mask, 11, 3.50, 41.195 
Mardi Gras Mask, 1, 3.50, 3.745 
Mardi Gras Mask, 1, 3.50, 3.745 
Mardi Gras Mask, 1, 3.50, 3.745 


test_core.py
#100% TEST COVERAGE FOR THE CORE
import core
# def core_test():
#     assert 
def test_sales_tax(sales_tax):
    assert sales_tax == 0.07
    True
def test_items():
    assert '1' == 'Mardi Gras Masks'
    True


    # def inventory():
#     str_1 = ['items, quantity, price']
#     inventory = []
#     for items in inventory:
#         if items[1] < 0:
#             items[1] = 1
#             items[1] = str(items[1])
#             str_1.append(', '.join(str_1))
#             message = '\n'.join(str_1)
#         with open('inventory.txt','w') as file:
#             file.write(message)
#         return inventory


# class something():
#     def get_item_prices(self):
#         """
#         """
#         party_inventory = []
#         if party_inventory == '1' or 'Mardi Gras Masks':
#             return 3.50
#         if party_inventory == '2' or 'Outdoor wooden dance floor':
#             return 50.00
#         if party_inventory == '3' or 'Can Lights':
#             return 2.50
#         if party_inventory == '4' or 'Mardi Gras Street Signs':
#             return 5.00
#         if party_inventory == '5' or 'Building Props':
#             return 25.00
#             print('Try Again.')
#             s = something
#             s.out()
#             return None

disk.py
def inventory_():
    with open('inventory.txt', 'r') as file:
        file.readline()
        items = file.readlines()
        for item in items:
            print(item)


def replacement_cost(num):
    """ (str) -> (int)
    Returns my replacement cost from
    inventory.txt
    """
    with open('inventory.txt', 'r') as file:
        file.readline()
        item = file.readlines()
    for i in item:
        if num in i:
            parts = i.split(', ')
            replacement_cost = parts[3].strip()
            return replacement_cost


def price(num):
    """(str) -> (int)
    Returns my price for the rent of
    the item from inventory.txt
    """
    with open('inventory.txt', 'r') as file:
        file.readline()
        item = file.readlines()
    for i in item:
        if num in i:
            parts = i.split(', ')
            price = parts[3].strip()
            return price


def update_history(item, days, rental_price, total):
    """ -> (txt)
    Will print text in history.txt of every transaction made.
    """
    transaction = '{}, {}, {}, {} \n'.format(item, days, rental_price, total)
    with open('history.txt', 'a') as file:
        file.write(transaction)


def update_inventory(item, quantity, price, replacement_cost):
    """ -> (txt)
    Will update when an item is purchased,
    and prints to inventory how many items
    are left
    """
    trans = '{}, {}, {}, {} \n'.format(quantity)
    with open('inventory.txt', 'a') as file:
        for quantity in quantities:
            if quantity <= 500:
                file.write(trans)
                return quantity

shell.py
#the built in functions input() and print() can only be called in the SHELL!!<<<>>>>
import core
import disk
import time,sys
from subprocess import check_output
#speak = check_output(['espeak','Welcome to Maegans New Orleans themed Party Rentals!'])
typing_speed = 12
def slow_type(t):
    for l in t:
        sys.stdout.write(l)
        sys.stdout.flush()
        time.sleep(typing_speed / 970.0)
    return input()
def main():
    print(slow_type(' 🎭𝓦𝓮𝓵𝓬𝓸𝓶𝓮 𝓽𝓸 𝓜𝓪𝓮𝓰𝓪𝓷𝓼 𝓝𝓮𝔀 𝓞𝓻𝓵𝓮𝓪𝓷𝓼 𝓣𝓱𝓮𝓶𝓮𝓭 𝓟𝓪𝓻𝓽𝔂 𝓡𝓮𝓷𝓽𝓪𝓵𝓼🎭 '))
    print(slow_type('𝓘 𝓱𝓸𝓹𝓮 𝔂𝓸𝓾 𝓯𝓲𝓷𝓭 𝔀𝓱𝓪𝓽 𝔂𝓸𝓾𝓻𝓮 𝓵𝓸𝓸𝓴𝓲𝓷𝓰 𝓯𝓸𝓻 ❗ 𝓗𝓪𝓿𝓮 𝓪 𝓖𝓻𝓮𝓪𝓽 𝓭𝓪𝔂! 🎉'))
    deposit = slow_type('𝓔𝓪𝓬𝓱  𝓬𝓾𝓼𝓽𝓸𝓶𝓮𝓻 𝔀𝓲𝓵𝓵  𝓱𝓪𝓿𝓮  𝓽𝓸  𝓹𝓪𝔂 𝓪 𝓭𝓮𝓹𝓸𝓼𝓲𝓽 𝓸𝓯 $5 𝓯𝓸𝓻 𝓮𝓪𝓬𝓱 𝓼𝓪𝓵𝓮')
    method = slow_type('𝓘𝓼 𝓽𝓱𝓲𝓼 𝓶𝓮𝓽𝓱𝓸𝓭 𝓸𝓴𝓪𝔂 𝔀𝓲𝓽𝓱 𝔂𝓸𝓾?(𝓨 \ 𝓝 )')
    if method == 'y':
        print(slow_type('\n𝓨𝓸𝓾𝓻 𝓶𝓸𝓷𝓮𝔂 𝔀𝓲𝓵𝓵 𝓫𝓮 𝓻𝓮𝓽𝓾𝓻𝓷𝓮𝓭 𝓸𝓷𝓬𝓮 𝓽𝓱𝓮 𝓹𝓻𝓸𝓭𝓾𝓬𝓽 𝓲𝓼 𝓲𝓷𝓼𝓹𝓮𝓬𝓽𝓮𝓭 𝓪𝓷𝓭 𝓷𝓸𝓽 𝓭𝓪𝓶𝓪𝓰𝓮𝓭\n'))
    elif method == 'n':
        print(slow_type('𝓢𝓸𝓻𝓻𝔂 𝓨𝓸𝓾 𝓬𝓪𝓷𝓽 𝓻𝓮𝓷𝓽 𝓪𝓷𝔂𝓽𝓱𝓲𝓷𝓰 𝓾𝓷𝓵𝓮𝓼𝓼 𝔂𝓸𝓾 𝓹𝓪𝔂 𝓽𝓱𝓮 𝓭𝓮𝓹𝓸𝓼𝓲𝓽.'))
        return None
    prices = ['::::\t1.𝓜𝓪𝓻𝓭𝓲 𝓖𝓻𝓪𝓼 𝓜𝓪𝓼𝓴𝓼 $3.50','::::\t2.𝓞𝓾𝓽𝓭𝓸𝓸𝓻 𝓦𝓸𝓸𝓭𝓮𝓷 𝓓𝓪𝓷𝓬𝓮 𝓕𝓵𝓸𝓸𝓻 $50.00', '::::\t3.𝓒𝓪𝓷 𝓵𝓲𝓰𝓱𝓽𝓼 $2.50','::::\t4.𝓜𝓪𝓻𝓭𝓲 𝓖𝓻𝓪𝓼 𝓢𝓽𝓻𝓮𝓮𝓽𝓼 𝓢𝓲𝓰𝓷𝓼 $5.00', '::::\t5.𝓑𝓾𝓲𝓵𝓭𝓲𝓷𝓰 𝓟𝓻𝓸𝓹𝓼 $25.00']
    print(*prices,sep='\n')
    purchase = slow_type('\n 𝓣𝓸 𝓶𝓪𝓴𝓮 𝓪 𝓼𝓮𝓵𝓮𝓬𝓽𝓲𝓸𝓷 𝓽𝔂𝓹𝓮 𝓲𝓷 𝓽𝓱𝓮 𝓷𝓾𝓶𝓫𝓮𝓻 𝓵𝓲𝓼𝓽𝓮𝓭 𝓫𝔂 𝓽𝓱𝓮 𝓲𝓽𝓮𝓶. \n')
    purchase = float(slow_type('\n 𝓔𝓷𝓽𝓮𝓻 𝓽𝓱𝓮 𝓹𝓻𝓲𝓬𝓮 𝓸𝓯 𝓽𝓱𝓮 𝓹𝓾𝓻𝓬𝓱𝓪𝓼𝓮: \n'))
    days = slow_type('\n 𝓗𝓸𝔀 𝓶𝓪𝓷𝔂 𝓭𝓪𝔂𝓼 𝔀𝓲𝓵𝓵 𝔂𝓸𝓾 𝓫𝓮 𝓻𝓮𝓷𝓽𝓲𝓷𝓰 𝓽𝓱𝓲𝓼 𝓹𝓻𝓸𝓭𝓾𝓬𝓽? \n')
    print(slow_type('\n 𝓣𝓱𝓪𝓷𝓴𝓼! 𝓟𝓪𝓻𝓽𝔂 𝓞𝓷....𝔀/𝓸𝓾𝓽 𝓽𝓱𝓮 𝓿𝓸𝓸𝓭𝓸𝓸 \n'))

    f = open('history.txt','r')
    if f.mode == 'r':
        contents = f.read()
        print (contents)
    f = open('inventory.txt','r')
    if f.mode == 'r':
        contents = f.read()
        print (contents)

    if purchase == 'restock':
        print(core.restock())
        return None
    if purchase == 'revenue':
        print('𝓨𝓸𝓾𝓻 𝓽𝓸𝓽𝓪𝓵 𝓻𝓮𝓿𝓮𝓷𝓾𝓮 𝓲𝓼 ${:.2f}'.format(core.revenue()))
        return None
   
    core.shopping_inven()
    core.calculate_totals(purchase)
if __name__ == '__main__':
    main()

shell2.py
import core2
import disk
import time, sys
typing_speed = 12


def slow_type(t):
    for l in t:
        sys.stdout.write(l)
        sys.stdout.flush()
        time.sleep(typing_speed / 970.0)
    return input()


def main():
    print('\n Welcome to Maegans New Oreleans Themed Party Rentals ')
    time.sleep(1.3)
    core2.options()

    choice = slow_type('\n What can I help you with today? \n')
    print('\n\nDisplaying inventory...')
    time.sleep(1)

    if choice == '1':
        disk.inventory_()
        rental_choice = slow_type('\nWhat item would you like to rent today? ')

        if rental_choice == '1':
            item = 'Mardi Gras Mask'
            days = slow_type(
                '\nHow many days would you like to rent this item? ')
            rental_price = disk.price(rental_choice)
            rental_replacement_cost = disk.replacement_cost(rental_choice)
            total = core2.total_cost(rental_price, days,
                                     rental_replacement_cost)
            print('Total Cost: $', total)

            disk.update_history(item, days, rental_price, total)

        if rental_choice == '2':
            item = 'Outdoor Wooden Dance Floor'
            days = slow_type(
                '\nHow many days would you like to rent this item? ')
            rental_price = disk.price(rental_choice)
            rental_replacement_cost = disk.replacement_cost(rental_choice)
            total = core2.total_cost(rental_price, days,
                                     rental_replacement_cost)
            print('Total Cost: $', total)

            disk.update_history(item, days, rental_price, total)

        if rental_choice == '3':
            item = 'Can Lights'
            days = slow_type(
                '\nHow many days would you like to rent this item? ')
            rental_price = disk.price(rental_choice)
            rental_replacement_cost = disk.replacement_cost(rental_choice)
            total = core2.total_cost(rental_price, days,
                                     rental_replacement_cost)
            print('Total Cost: $', total)

            disk.update_history(item, days, rental_price, total)

        if rental_choice == '4':
            item = 'Mardi Gras Street Signs'
            days = slow_type(
                '\nHow many days would you like to rent this item? ')
            rental_price = disk.price(rental_choice)
            rental_replacement_cost = disk.replacement_cost(rental_choice)
            total = core2.total_cost(rental_price, days,
                                     rental_replacement_cost)
            print('Total Cost: $', total)

            disk.update_history(item, days, rental_price, total)

        if rental_choice == '5':
            item = 'Building Props'
            days = slow_type(
                '\nHow many days would you like to rent this item? ')
            rental_price = disk.price(rental_choice)
            rental_replacement_cost = disk.replacement_cost(rental_choice)
            total = core2.total_cost(rental_price, days,
                                     rental_replacement_cost)
            print('Total Cost: $', total)

            disk.update_history(item, days, rental_price, total)
            disk.update_inventory(trans)


if __name__ == '__main__':
    main()

core.py
#functions go in the core
def calculate_totals(purchase):
    days = 365
    prices = 0.00
    replacement_value = .10
    deposit = 5.00
    sales_tax = purchase * .07
    total_price = purchase + sales_tax + deposit + replacement_value
    print('\n 𝓣𝓸𝓽𝓪𝓵 𝓒𝓸𝓼𝓽: '+ '$' + str(total_price) + ".",sep="" + '\n')

def shopping_inven():
    stock = {
        "Mardi Gras Masks":500, 
        "Outdoor Wooden Dance Floor":500, 
        "Can Lights":500,
        "Mardi Gras Street Signs":500, 
        "Building Props":500
        }
    prices = {
        "Mardi Gras Masks":3.50,
        "Outdoor Wooden Dance Floor":50.00,
        "Can Lights":2.50,
        "Mardi Gras Street Signs":5.00,
        "Building Props":25.00
    }

def compute_bill(items):
    total = 0 
    if stock[items] > 0:
            total += prices[items]
            stock[items] -= 1    
    return total